pkg/v1/tarball/write.go

```diff
@@ -66,6 +72,16 @@ func Write(tag name.Tag, img v1.Image, wo *WriteOptions, w io.Writer) error {
 		return err
 	}
 
+	// Default filter function
+	filter := func(_ v1.Layer) (bool, error) {
+		return true, nil
+	}
+	if wo != nil {
+		if wo.LayerFilter != nil {
```

Comment:
Collapse this into `if wo != nil && wo.LayerFilter != nil {` ?

Though I guess that makes it harder to add more options... up to you.